orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/ExecutionMapper.kt

```diff
@@ -75,7 +77,24 @@ class ExecutionMapper(
         }
 
         executions.forEach { execution ->
-          execution.stages.sortBy { it.refId }
+          val stages = mutableListOf<Stage>()
+
+          stages.addAll(
+            execution.stages.filter { it.parentStageId == null }
+              .sortedBy { it.refId }
+          )
+
+          execution.stages.filter { it.parentStageId != null }
+            .sortedBy { it.refId }
+            .forEach {
+              when (it.syntheticStageOwner) {
+                STAGE_BEFORE -> stages.add(stages.indexOf(it.parent), it)
+                STAGE_AFTER -> stages.add(stages.indexOf(it.parent) + 1, it)
```

Comment:
Ensures `SqlExecutionRepository` returns executions with stages sorted in the same order as `RedisExecutionRepository`. Instead of sorting all stages by `refId`, synthetic `BEFORE` and `AFTER` stages are clustered around their parent.